compiler/src/org.graalvm.compiler.core/src/org/graalvm/compiler/core/gen/NodeLIRBuilder.java

```diff
@@ -226,6 +229,19 @@ public void setMatchResult(Node x, Value operand) {
         nodeOperands.set(x, operand);
     }
 
+    public void setSharedMatchResult(Node node) {
+        assert nodeOperands != null && nodeOperands.get(node) == null : "operand cannot be set twice";
+        SharedMatchValue matchValue = sharedMatchValues.get(node);
```

Comment:
You know I don't really thing we need SharedMatchValue.  If we're going to maintain a side table we can just as easily store an Integer to track the count.  Then when it's equal to getUsageCount() we can store INTERIOR_MATCH into nodeOperands.

Reply:
Yeah, agree! Actually the additional `SharedMatchValue` is not needed, and I think it's ok to remove it. Thanks for your review! And it's great that you could help to make it better. Thanks so much!